### ISSUE TITLE
RsaOpenSsl `TryDecrypt` does not behave the same manner than `Decrypt` 

### ISSUE THREAD

**Issue by ycrumeyrolle** [dotnet/runtime#38767]
The following test is working on Windows, but does not work on Linux. 
The encryption is fine, the decryption with `Decrypt` is also woking. 
The `TryDecrypt` method returns `false` since the destination buffer is of the correct size.

Tried on SDK 3.1.301, Ubuntu 18.04, in netcoreapp2.1.

```
using System;
using System.Globalization;
using System.Security.Cryptography;
using System.Text;
using Xunit;

public class RsaTests
{
    [Fact]
    public void EncryptThenDecrypt_SameoutputThanInput()
    {
        var parameters = new RSAParameters
        {
            Exponent = "010001".HexToByteArray(),
            Modulus = "F34982DF8C13A7F0179A579578E5D8BE4F222C57BCEE8B6ED0A2DD2C4F02F9856F216BB46109EDB856EA53FF065061104C90527BDE2DB1EE724D1F8EAC737FA265B91DFCB433AA0840B9D2EA6D9C9B758DF13A195727BB98CE40B43E6346D30CA1BFC2F383F50F626913885FD49AD7245464C386F2D645D8D06234FA9E4B856F362163E7C9AE545C4A2A20FE1937DB81D0ADEAE0A21BDF8A61094EB815094B51AB3AB328A91BE593A1C33CCF439D946C0060A96624F9664A68D4ACE00493106CF5FAB9E3DAB677E16CD788B71B0FB5610C5E125F2084D786DC15568C9B8A87F64D2799AB3D120CC9D56EE00D1563382BE99C35AB759FE32DAA75DA0BF9A62B91".HexToByteArray(),
            D = "57B099718605D7B45C6D8F401B118DAC1ACDD079D7698BEC675B2CA939D972A2BBC9AA0E9C237E01C1A238A06560377C2E7C9E9E7318B9579EEF597E8D7E1BF527013063D37803C371C108DAE2A35E0D631A1DBCE0833BC4FA22AC55086790CEAEC675B85C49CA3FC86BF7EDC2AE04F9DD496C8889A2DE0E261E1FC66DA3FEF5C0F5F65A8644F90CB65E504B215830E0F04B8FDDC4E463A580B0F9CBF4CA137C222702E7F4E45BCE8F4940B7E220C09913A045599570EB35EDF95053DFD1B4ED4DBE68419662C536C7F036EDD7DBFBE64EE9ED494AC7FFBE83213F646B7461183E444BEB9EC456E76B7952503D5F7F7E3D23A5F029C45AEB70D6094D8BCCB459".HexToByteArray(),
            DP = "424EAC1C20150733EB7545377C6F2C4928D35C4574EA12557C09327649125E8486063A9A6EA9162F3B458208BF6838EC30BD7C82F72CC4B9FBE09C46720ABC9E7343BC26F8390DB2FA9DA7F457E369969171905281F42EDC9595C267F78CF489015B8932836CD88244164088B1F433DDC2B138680BDEBE8BB13BAC2C38392E0D".HexToByteArray(),
            DQ = "37EB3A4F2CE0C01690DA4AF7AFADCFF0ED580805279EB89D6B1700F8ECAD33F27E2FB63172CEF2A5C0CB3D84B6FDC2A1D0ACAC095C64BCDFA2D9BF19FCB0E1FFEC8150F910302AA7DCC07F84F4B57A623CC583F1170197AE2A3B9BDB2E925310181752BC68C31FA7431151238A1F196726A5B55CCEEDE5A2453F99A3E0870E69".HexToByteArray(),
            P = "F4A7F901C2D946E551294256AC6A51B24D60176B0BE5EE4B711F14C699FEE611E13240856C4A7098E970B2B187599F9CFAB101D290FD25149E7679B020FE566FAB2B42405817605D48BC70E8E88FB01FC07866AD81524F0D01BC07763DEE3EA32E199387B0E4B3834F2239940E23318595F6FE514AC8F2FA252D0459E245ACE7".HexToByteArray(),
            Q = "FE9149F29CE9D3087B36C47FA7E53F7F58DC08337E0AAC7EADC7BC9110AEB560D8D899A9CEA468C9C9D0E1A0568A05D33F10F1194254CF7ED24995832960C92C254B6CCD297DBD233301DAAEA15E3B75D94CE7B835D2985BA68C85471A585FA494F00BBB598BA3253FF4FA778ECE7C0568269FCE000C50CEC1EF9EC13AE09CC7".HexToByteArray(),
            InverseQ = "AA30B26405C977F81ADC758E962E027A37B7BE70EFDB0C2A7ECE007D3F87628BBD33F1BD01F2F1C76094B0B2FB3FC636944854F8BEAF4625321EA1B3E44903D435127677EA4F42C09DD730D849CBA285F519C3EA09B8BE0708F188F885E73D9F9708D8410437CA876FF93A130F1CB76FD855ABE5B594B3E57E1AC923070406EA".HexToByteArray()
        };
        using RSA rsa = RSA.Create();
        rsa.ImportParameters(parameters);

        var data = "F8F61435DBCB5D59DCF32250C3B89091294D9806F6D3664EFAB7C2CAB01CD4BA".HexToByteArray();
        var encryptedData = rsa.Encrypt(data, RSAEncryptionPadding.Pkcs1);
        var decryptedData = rsa.Decrypt(encryptedData, RSAEncryptionPadding.Pkcs1);
        
        Assert.Equal("F8F61435DBCB5D59DCF32250C3B89091294D9806F6D3664EFAB7C2CAB01CD4BA", decryptedData.ByteArrayToHex());

        CryptographicOperations.ZeroMemory(decryptedData);
        var decrypted = rsa.TryDecrypt(encryptedData, decryptedData, RSAEncryptionPadding.Pkcs1, out int bytesWritten);

        Assert.Equal("F8F61435DBCB5D59DCF32250C3B89091294D9806F6D3664EFAB7C2CAB01CD4BA", decryptedData.ByteArrayToHex());
        Assert.True(decrypted);
        Assert.Equal(32, bytesWritten);
    }
}

internal static class ByteUtils
{
    internal static byte[] HexToByteArray(this string hexString)
    {
        byte[] bytes = new byte[hexString.Length / 2];

        for (int i = 0; i < hexString.Length; i += 2)
        {
            string s = hexString.Substring(i, 2);
            bytes[i / 2] = byte.Parse(s, NumberStyles.HexNumber, null);
        }

        return bytes;
    }

    internal static string ByteArrayToHex(this byte[] bytes)
    {
        StringBuilder builder = new StringBuilder(bytes.Length * 2);

        for (int i = 0; i < bytes.Length; i++)
        {
            builder.Append(bytes[i].ToString("X2"));
        }

        return builder.ToString();
    }
}
```

**Comment by vcsjones** [comment]
A few things.

First, the code sample as-is does not compile. When I fixed the compilation error, I was not able to reproduce this in net50 or in netcoreapp3.1 on Ubuntu 20.04.

> Tried on SDK 3.1.301

What is your `<TargetFramework>` in the project? 

This seems to be the same issue that was reported in #27942, which was fixed in https://github.com/dotnet/corefx/pull/36601.

**Comment by ycrumeyrolle** [comment]
> First, the code sample as-is does not compile 

Fixed
 
> What is your `<TargetFramework>` in the project?
netcoreapp2.1 & netcoreapp3.0
 
> This seems to be the same issue that was reported in #27942, which was fixed in [dotnet/corefx#36601](https://github.com/dotnet/corefx/pull/36601), so it should be fixed when the `TargetFramework` is netcoreapp3.1 or higher.

**Comment by bartonjs** [comment]
Closing since this is something we've already fixed.
